linkerd/app/inbound/src/direct.rs

```diff
@@ -137,20 +135,46 @@ impl<N> Inbound<N> {
                                 TransportHeader {
                                     port,
                                     name: None,
-                                    protocol: None,
+                                    protocol,
                                 } => {
                                     // When the transport header targets an alternate port (but does
                                     // not identify an alternate target name), we check the new
                                     // target's policy to determine whether the client can access
                                     // it.
                                     let addr = (client.local_addr.ip(), port).into();
-                                    let allow = policies.check_policy(OrigDstAddr(addr))?;
-                                    Ok(svc::Either::A(Local {
-                                        addr: Remote(ServerAddr(addr)),
-                                        policy: allow,
-                                        protocol: None,
-                                        client,
-                                    }))
+                                    let policy = policies.check_policy(OrigDstAddr(addr))?;
+                                    let local = match protocol {
+                                        None => {
+                                            let tls = tls::ConditionalServerTls::Some(
+                                                tls::ServerTls::Established {
+                                                    client_id: Some(client.client_id.clone()),
+                                                    negotiated_protocol: client.alpn,
+                                                },
+                                            );
+                                            let permit = policy
```

Comment:
Ah nice, so when we go direct w/o a protocol we can just check the policy here. This is something I was wondering about in my approach; if we check the policy in the direct stack, it'll be checked once again in the http stack?

Reply:
yeah, in this case we only check the authorization for the tcp stack -- this is why the LocalTcp type takes a `permit`: to prove that it has been authorized.

the http stack will do its own authorization checks on requests.